lib/ansible/modules/cloud/docker/docker_container.py

```diff
@@ -1328,12 +1330,12 @@ def __init__(self, client):
 
         if self.entrypoint:
             # convert from list to str.
-            self.entrypoint = ' '.join([str(x) for x in self.entrypoint])
+            self.entrypoint = ' '.join([to_text(x, errors='surrogate_or_strict') for x in self.entrypoint])
```

Comment:
Is there a reason `to_text` is being used here. The rule is that text strings should only be used on the module side when dealing with an API that requires text strings.

In a module native should be used everywhere, except in special cases. This is in opposition to controller code, where text should be used everywhere, except in special cases.

https://gist.github.com/sivel/9eb368f98974e15033c3ac2f989ac4e0

Reply:
Docker (and previously docker-py) allow both `str` and `unicode` with Python 2, but everything they return from the docker daemon is `unicode`. The module fails miserably because it compares (in a sort operation) data returned by the daemon (`unicode`) with user input (`str`), and that fails. Thus we either change the module to convert everything the daemon gives to `str` (using `to_native`), or we convert all user input to `unicode` (using `to_text`). I think converting everything to `unicode` is better in this case.